cli/remoteforward.go

```diff
@@ -0,0 +1,104 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"regexp"
+	"strconv"
+
+	gossh "golang.org/x/crypto/ssh"
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/agent/agentssh"
+)
+
+// cookieAddr is a special net.Addr accepted by sshRemoteForward() which includes a
+// cookie which is written to the connection before forwarding.
+type cookieAddr struct {
+	net.Addr
+	cookie []byte
+}
+
+var remoteForwardRegex = regexp.MustCompile(`^(\d+):(.+):(\d+)$`)
+
+func validateRemoteForward(flag string) bool {
+	return remoteForwardRegex.MatchString(flag)
+}
```

Comment:
nit: why not just try `parseRemoteForward` and check if err != nil instead?

Reply:
This way I don't need to expose `localAddr`, `remoteAddr`, and some bool flag to see if remote-forward mode is enabled. I can adjust, but would not like to have too many vars exposed.